doc/source/serve/autoscaling.md

```diff
@@ -0,0 +1,43 @@
+(serve-autoscaling)=
+
+# Serve Autoscaling
+
+This section should help you:
+
+- Understand how Ray Serve autoscaling works
+- Understand config parameter usage
+
+
+## Autoscaling architecture
+Ray Serve autoscaling is to increase and decrease the number of replicas based on the load, and ray serve provide a couple of config parameters to meet the different workload use cases.
+
+![pic](https://raw.githubusercontent.com/ray-project/images/master/docs/serve/autoscaling.svg)
+
+- Deployment handle(Client) and worker replica periodically push the stats to the autoscaler.
+- Autoscaler requires serve handle queue metrics and replicas requests metrics to make decision whether to scale (up or down) the replicas.
+- Deployment Handle(Client) keeps polling the replica stats from controller to get the updated replicas information. Serve Handle(Client) will send requests directly to the replica based on the replicas information (Round Robin).
+
+:::{note}
+When the controller dies, the client is still able to send requests, but autoscaling will not work.
```

Comment:
autoscaling will pause to work, when controller recovers, autoscaling will start afresh. All the previous metrics collected will be lost.